### PR TITLE
Update to use napari-sphinx-theme 0.3.0

### PR DESCRIPTION
**Pull request by melissawm**
Prepares the docs repo to use the new napari-sphinx-theme, 0.3.0, which depends on pydata-sphinx-theme rather than forking it.

# References and relevant issues

Addresses napari/napari-sphinx-theme#113
Depends on napari/napari-sphinx-theme#134

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
This will also require the constraints -- which are in napari/napari -- to get updated once the new theme is released.

**Comment by Czaki** [comment]
Is the new theme is incompatible with the old one and could break 0.4.19 build docs process?

**Comment by jni** [comment]
@Czaki is the theme not part of the constraints?

**Comment by Czaki** [comment]
> @Czaki is the theme not part of the constraints?

Is. but we do not have upper constraints at this moment. So I would like to know if I should add it in the 0.4.19 branch.

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
> Is. but we do not have upper constraints at this moment. So I would like to know if I should add it in the 0.4.19 branch.

I thought constraints were pins? That's what I was asking anyway.

Maybe this PR could get backported? It seems unobtrusive enough... Can't wait to have the new theme! But up to you. We can wait to release the theme until after 0.4.19.

**Comment by Czaki** [comment]
> I thought constraints were pins? That's what I was asking anyway.
> 
> Maybe this PR could get backported? It seems unobtrusive enough... Can't wait to have the new theme! But up to you. We can wait to release the theme until after 0.4.19.

I have a plan to refresh constraints before release 0.4.19. If a new theme is released earlier such a refresh will bring it in constraints.

**Comment by psobolewskiPhD** [comment]
This was discussed at the docs working group meeting today. The consensus was to wait until after 0.4.19 release, with the goal of the next release being 0.5.0 and using the split repo setup rather than cherry picking and getting a fresh look!

**Comment by melissawm** [comment]
This is rebased and should be ready to go along with napari/napari-sphinx-theme#134

**Comment by jni** [comment]
@melissawm the CI disagrees... 😅 Perhaps you can point it to your branch for now, with a comment that it needs to be updated to depend on `napari-sphinx-theme` directly once that branch is released?

**Comment by melissawm** [comment]
Yeah I was assuming this would only go in after the theme was updated, but I can do this for the preview.

**Comment by melissawm** [comment]
Ok it looks like it's alive - you can see it here https://output.circle-artifacts.com/output/job/c9c574d3-71c6-434f-b5ab-85bdb7ff68c3/artifacts/0/docs/docs/_build/index.html?pr=267 🥳 

(note the calendar events are not showing up since we don't use the calendar API key in the circleci set up - I can fix that in a separate commit. The deploy does have the correct env variable set up though.)

**Comment by melissawm** [comment]
Ok, here's a new attempt! This time I decided to try my hand at javascript (just a little 😅 )

How does that look? https://melissawm.github.io/napari-theme-var/community/meeting_schedule.html

**Comment by psobolewskiPhD** [comment]
Looks awesome to me!
And I love that the links just work!
:tada:

**Comment by psobolewskiPhD** [comment]
It's a bit annoying the docs pip constraints are in the other repo.
I think we will need to:
1) merge the theme pr
2) merge this pr
3) checkout main napari, install pip-tools run:
`pip-compile --allow-unsafe --extra=optional --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --extra=testing_extra --output-file=napari_repo/resources/constraints/constraints_py3.10_docs.txt --strip-extras docs/requirements.txt napari_repo/resources/constraints/pydantic_le_2.txt napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg resources/constraints/version_denylist_examples.txt`
and make a PR to napari/napari with that constraints file.
Is that correct @Czaki ?

Edit: except that `napari_repo` and `docs` need to be the proper names of the local repos.

File: requirements.txt [hunk 1]
```diff
@@ -9,7 +9,8 @@ sphinx-copybutton
 sphinx-gallery
 sphinx_autodoc_typehints==1.12.0
 myst-nb
-napari-sphinx-theme
+#napari-sphinx-theme>=0.3.0
```
Comment (melissawm) [inline requirements.txt]: Let me fix this before we merge

**Comment by melissawm** [comment]
Please squash before merging 😄

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
Alas, I can't get the pip-tools thing to work (testing on current stuff) because I'm on arm64 and there are missing wheels (triangle, pyside2, maybe more). I guess it should be run under linux, like CI?
I don't have a VM handy at the moment.

**Comment by psobolewskiPhD** [comment]
Oh and we need a release of the theme!

**Comment by melissawm** [comment]
@jni has cut a 0.3.0 release of the theme so we're good there 🎉 @psobolewskiPhD I'm on linux so if you point me to the instructions I might be able to update the constraints, let me know!

**Comment by jni** [comment]
Correction: @jni *attempted* to push a 0.3.0 release, but the workflow failed. 😂 See https://pypi.org/project/napari-sphinx-theme/ and napari/napari-sphinx-theme#135.

**Comment by jni** [comment]
For anyone coming across this and related PRs, sorry about the fast merges, but @melissawm, @psobolewskiPhD, and I happened to coincide sync and we worked together on merging all of these in sequence, forgoing protocol, since they all depend on each other and merging some but not others could result in broken builds and other inconsistencies. The related PRs are:

- napari/napari-sphinx-theme#134
- napari/napari-sphinx-theme#135
- napari/napari-sphinx-theme#136
- napari/napari-sphinx-theme#137
- napari/napari#6598
- napari/docs#267

You can relive all the action in the [docs-wg thread on Zulip](https://napari.zulipchat.com/#narrow/stream/298358-working-group-documentation/topic/meeting.202024-01-17/near/416464138)! We now resume our regularly scheduled programming. But check out the new docs! 😍 https://napari.org/dev (new on left, old on right)

image

**Pull request merged**